### ISSUE TITLE
`cancel` method does not finish procedure.

### ISSUE THREAD

**Issue by tapoton** [ProcedureKit/ProcedureKit#705]
After update to 4.1.0 `cancel` method does not finish procedure anymore.

Was that done on purpose? If yes, how am I supposed to handle it using 4.1.0? Should I call `finish` manually?

**Comment by swiftlyfalling** [comment]
Yes, this was indeed a change in Beta 7 (and it looks like the ChangeLog ought to be updated with more details) - here is the PR: https://github.com/ProcedureKit/ProcedureKit/pull/571 (Please note: The PR contains draft documentation that is now outdated. See below.)

It was determined that it is not safe for the framework to automatically finish a Procedure subclass after the Procedure subclass's `execute()` has been called (since the framework has no idea what your code / subclass is doing), so the behavior was changed.

Thus, it is up to your Procedure subclass to handle calling `finish()` as appropriate in *all* cases (including cancellation).

I've just updated the documentation on this in the Wiki. Please see here for a review of the options (with examples): https://github.com/ProcedureKit/ProcedureKit/wiki/Handling-Cancellation
